source/agora/consensus/EnrollmentManager.d

```diff
@@ -449,7 +449,11 @@ public class EnrollmentManager
         const enrolled_height =
             this.validator_set.getEnrolledHeight(this.enroll_key);
         // FIXME: This function should only be called if we're enrolled
-        assert(enrolled_height != ulong.max);
+        if (enrolled_height == ulong.max)
```

Comment:
That's a good change, but what about a unittest?

Reply:
> That's a good change, but what about a unittest?

I added the test.